src/renderer/components/database-list-item-metadata.jsx

```diff
@@ -33,10 +47,60 @@ export default class DbMetadataList extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if ((this.props.columnsByTable !== nextProps.columnsByTable) ||
+    (this.state.tableheights !== nextState.tableheights) ||
+    this.recalc === true ||
+    nextProps.items !== this.props.items ||
+    this.state.collapsed !== nextState.collapsed) {
+      this.recalc = true;
+      return true;
+    }
+    return false;
+  }
+
+  componentDidUpdate() {
+    if (this.recalc === true) {
+      this.recalc = false;
+      this.metadataScroll.recomputeRowHeights();
+      this.metadataScroll.forceUpdate();
+    }
+  }
+
+  onAdjustHeight({ height, table, showcolumns, showtriggers, tableCollapsed }) {
+    const temptableheights = this.state.tableheights;
+    const tempshowcolumns = this.state.showcolumns;
+    const tempshowtriggers = this.state.showtriggers;
+    const temptablecollapsed = this.state.tableCollapsed;
+    if (temptableheights[table] !== height) {
+      temptableheights[table] = height;
+      tempshowcolumns[table] = showcolumns;
+      tempshowtriggers[table] = showtriggers;
+      temptablecollapsed[table] = tableCollapsed;
```

Comment:
These all are changing the `this.state` directly (even trough a new variable). With ES2015 you could do something like this:
```js
this.setState({
  temptableheights[table]: height,
  tempshowcolumns[table]: showcolumns,
});
```